hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DeserializedGenericRecordBuilder.java

```diff
@@ -32,27 +32,30 @@
  */
 public class DeserializedGenericRecordBuilder extends AbstractGenericRecordBuilder {
 
-    private final TreeMap<String, Object> objects = new TreeMap<>();
+    private final TreeMap<String, Object> objects;
     private final SchemaWriter schemaWriter;
+    private boolean built;
 
     public DeserializedGenericRecordBuilder(String typeName) {
+        this.objects = new TreeMap<>();
         schemaWriter = new SchemaWriter(typeName);
     }
 
     /**
      * @return newly created GenericRecord
-     * @throws HazelcastSerializationException if a field is not written when building with builder from
-     *                                         {@link GenericRecordBuilder#compact(String)} (ClassDefinition)} and
-     *                                         {@link GenericRecord#newBuilder()}
      */
     @Nonnull
     @Override
     public GenericRecord build() {
+        this.built = true;
         return new DeserializedGenericRecord(schemaWriter.build(), objects);
     }
 
 
     protected GenericRecordBuilder write(@Nonnull String fieldName, Object value, FieldKind fieldKind) {
+        if (this.built) {
+            throw new HazelcastSerializationException("Cannot modify the GenericRecordBuilder after building");
```

Comment:
This is an error in the method argument. Should the error be `IllegalArgumentException` for such cases?

Reply:
I don't think it is related to method argument. Can you explain? This error is thrown if genericRecordBuilder.setXX() is called after genericRecordBuilder.build() is called. So it is more of an unsupported operation exception.

Reply:
`UnsupportedOperationException` is fine.